libgadget/timestep.c

```diff
@@ -188,32 +200,45 @@ find_timesteps(const ActiveParticles * act, DriftKickTimes * times)
         #pragma omp parallel for reduction(min:dti_min)
         for(i = 0; i < PartManager->NumPart; i++)
         {
+            enum TimeStepType titype;
             /* Because we don't GC on short timesteps, there can be garbage here.
              * Avoid making it active. */
             if(P[i].IsGarbage || P[i].Swallowed)
                 continue;
-            inttime_t dti = get_timestep_ti(i, dti_max, times->Ti_Current);
+            inttime_t dti = get_timestep_ti(i, dti_max, times->Ti_Current, &titype);
             if(dti < dti_min)
                 dti_min = dti;
         }
         MPI_Allreduce(MPI_IN_PLACE, &dti_min, 1, MPI_INT, MPI_MIN, MPI_COMM_WORLD);
     }
 
+    int64_t ntiaccel=0, nticourant=0, ntiaccrete=0, ntineighbour=0, ntihsml=0;
     int badstepsizecount = 0;
     int mTimeBin = TIMEBINS, maxTimeBin = 0;
-    #pragma omp parallel for reduction(min: mTimeBin) reduction(+: badstepsizecount) reduction(max:maxTimeBin)
+    #pragma omp parallel for reduction(min: mTimeBin) reduction(+: badstepsizecount, ntiaccel, nticourant, ntiaccrete, ntineighbour, ntihsml) reduction(max:maxTimeBin)
     for(pa = 0; pa < act->NumActiveParticle; pa++)
     {
         const int i = get_active_particle(act, pa);
 
         if(P[i].IsGarbage || P[i].Swallowed)
             continue;
 
+        enum TimeStepType titype;
         inttime_t dti;
         if(TimestepParams.ForceEqualTimesteps) {
             dti = dti_min;
         } else {
-            dti = get_timestep_ti(i, dti_max, times->Ti_Current);
+            dti = get_timestep_ti(i, dti_max, times->Ti_Current, &titype);
+            if(titype == TI_ACCEL)
```

Comment:
switch () {

case:

default:
raise(...)
}

Reply:
I could but it is longer code with a switch: easier to leave it as an if chain